backend/src/hmc-api/fetcher/types.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from "zod";
+import type { CourseFiles } from "../course";
+import { TermIdentifierString } from "hyperschedule-shared/api/v4";
+
+/**
+ * defines which param an endpoint use, if any
+ */
+export const Param = z.union([
+    z.literal("catalog"),
+    z.literal("session"),
+    z.literal("year"),
+]);
+export type Param = z.infer<typeof Param>;
+
+export const Params = z.object({
+    catalog: z.string().regex(/UG\d{2}/),
+    session: TermIdentifierString,
+    year: z.string().regex(/^20\d{2}$/),
+});
+export type Params = z.infer<typeof Params>;
+
+const Endpoint = z.object({
+    link: z.string().url(),
+    param: Param.nullable(),
```

Comment:
can you explain how this is supposed to work, since it looks like currently all the endpoints are defined with `null` param?